ip4defrag/defrag.go

```diff
@@ -188,15 +198,20 @@ type fragmentList struct {
 // See: http://www.sans.org/reading-room/whitepapers/detection/ip-fragment-reassembly-scapy-33969
 func (f *fragmentList) insert(in *layers.IPv4) (*layers.IPv4, error) {
 	// TODO: should keep a copy of *in in the list
-	// or not (ie the packet source is reliable) ?
+	// or not (ie the packet source is reliable) ? -> depends on Lazy / last packet
 	fragOffset := in.FragOffset * 8
 	if fragOffset >= f.Highest {
 		f.List.PushBack(in)
 	} else {
 		for e := f.List.Front(); e != nil; e = e.Next() {
 			frag, _ := e.Value.(*layers.IPv4)
-			if in.FragOffset <= frag.FragOffset {
-				debug.Printf("defrag: inserting frag %d before existing frag %d \n",
+			if in.FragOffset == frag.FragOffset {
```

Comment:
It's important to note that I'm not talking about overlaps here.  I'm talking about two packets with the same initial data, one of which is longer than the other.  Note that BB is the same in both, so regardless of the overlap strategy used (figure 2:5 in the linked doc), the "correct" output is AAAABBCCDDDD.  I just worry that this code will result in AAAABBxxDDDD, where x is "missing and unrecoverable".

Also, is your "which is a big bug" pointing out a bug in the current code, or a fix in that code?  Sorry, I'm just unclear.  Line number?